kernel/shared/src/main/scala/cats/effect/kernel/Resource.scala

```diff
@@ -484,6 +484,178 @@ sealed abstract class Resource[F[_], +A] {
+   * 3. If the fiber succeeds and .cancel won the race, finalize eagerly and
+   *    `join` results in `Canceled()`
+   * 4. If the fiber succeeds and .cancel lost the race or wasn't called,
+   *    finalize naturally when the containing scope ends, `join` returns
+   *    the value
+   */
+  def start[B >: A](
+      implicit F: Concurrent[F]): Resource[F, Fiber[Resource[F, *], Throwable, B]] = {
+    final case class State(
+        fin: F[Unit] = F.unit,
+        finalizeOnComplete: Boolean = false,
+        confirmedFinalizeOnComplete: Boolean = false)
+
+    Resource {
+      import Outcome._
+
+      F.ref[State](State()) flatMap { state =>
+        val finalized: F[B] = F uncancelable { poll =>
+          poll(this.allocated) guarantee {
+            // confirm that we completed and we were asked to clean up
+            // note that this will run even if the inner effect short-circuited
+            state update { s =>
+              if (s.finalizeOnComplete)
+                s.copy(confirmedFinalizeOnComplete = true)
+              else
+                s
+            }
+          } flatMap {
+            // if the inner F has a zero, we lose the finalizers, but there's no avoiding that
+            case (a, rel) =>
+              val action = state modify { s =>
+                if (s.confirmedFinalizeOnComplete)
+                  (s, rel.handleError(_ => ()))
+                else
+                  (s.copy(fin = rel), F.unit)
+              }
+
+              action.flatten.as(a)
+          }
+        }
+
+        F.start(finalized) map { outer =>
+          val fiber = new Fiber[Resource[F, *], Throwable, B] {
+            def cancel =
+              Resource eval {
+                F uncancelable { poll =>
+                  // technically cancel is uncancelable, but separation of concerns and what not
+                  poll(outer.cancel) *> state.update(_.copy(finalizeOnComplete = true))
+                }
+              }
+
+            def join =
+              Resource eval {
+                outer.join.flatMap[Outcome[Resource[F, *], Throwable, B]] {
+                  case Canceled() =>
+                    Outcome.canceled[Resource[F, *], Throwable, B].pure[F]
+
+                  case Errored(e) =>
+                    Outcome.errored[Resource[F, *], Throwable, B](e).pure[F]
+
+                  case Succeeded(fp) =>
+                    state.get map { s =>
+                      if (s.confirmedFinalizeOnComplete)
+                        Outcome.canceled[Resource[F, *], Throwable, B]
+                      else
+                        Outcome.succeeded(Resource.eval(fp))
+                    }
+                }
+              }
+          }
+
+          val finalizeOuter =
+            state.modify(s => (s.copy(finalizeOnComplete = true), s.fin)).flatten
+
+          (fiber, finalizeOuter)
+        }
+      }
+    }
+  }
+
+  def evalOn[B >: A](ec: ExecutionContext)(implicit F: Async[F]): Resource[F, B] =
```

Comment:
```suggestion
  def evalOn(ec: ExecutionContext)(implicit F: Async[F]): Resource[F, A] =
```

Another unnecessary context bound.